lib/document.js

```diff
@@ -2524,9 +2524,8 @@ Document.prototype.$__validate = function(pathsToValidate, options, callback) {
       // If you `populate()` and get back a null value, required validators
       // shouldn't fail (gh-8018). We should always fall back to the populated
       // value.
-      let pop;
-      if (val == null && (pop = _this.populated(path))) {
-        val = pop;
+      if (val == null && _this.populated(path)) {
```

Comment:
We should remove this `val == null` check. This `null` check is the crux of this issue: we should always use `val = pop`